src/core_modules/capture-core/components/Pages/Enrollment/EnrollmentPageDefault/runRulesForEnrollment.js

```diff
@@ -0,0 +1,109 @@
+            });
+            return acc;
+        }, {});
+};
+
+const getEventValuesFromEvent = (enrollment, eventId, dataElements) => {
+    const currentEvent = enrollment.events.find(item => item.event === eventId);
+    const eventValues = {};
+    if (currentEvent) {
+        currentEvent.dataValues.forEach((dataValue) => {
+            const { dataElement } = dataValue;
+            eventValues[dataElement] = dataElements[dataElement].value;
+        });
+    }
+
+    return eventValues;
+};
+
+
+const getEventsDataFromEnrollment = (enrollment, dataElements) => {
+    const allEvents = enrollment.events.map(event => ({
+        eventId: event.event,
+        programId: event.program,
+        programStageId: event.programStage,
+        orgUnitId: event.orgUnit,
+        orgUnitName: event.orgUnitName,
+        trackedEntityInstanceId: event.trackedEntityInstance,
+        enrollmentId: event.enrollment,
+        enrollmentStatus: event.enrollmentStatus,
+        status: event.status,
+        eventDate: convertDate(event.dueDate),
+        dueDate: convertDate(event.dueDate),
+        ...getEventValuesFromEvent(enrollment, event.event, dataElements),
+    }));
+
+    const groupByProgramStageId = allEvents.reduce((acc, currentEvent) => {
+        acc[currentEvent.programStageId] = { ...currentEvent };
+        return acc;
+    }, {});
+
+    return { all: allEvents, byStage: groupByProgramStageId };
+};
+
+
+const getEnrollmentData = enrollment => ({
+    enrollmentDate: convertDate(enrollment.enrollmentDate),
+    incidentDate: convertDate(enrollment.incidentDate),
+    enrollmentId: enrollment.enrollment,
+});
+
+const flatDataValuesFromEvents = events => events.reduce((acc, currentEvent) => {
+    acc = [...acc, ...(currentEvent.dataValues)];
+    return acc;
+}, []);
+
+export const runRulesForEnrollment = (input: InputRuleEnrollmentData) => {
+    const { orgUnit, program, programMetadata, enrollment, attributes } = input;
+    if (orgUnit && program && programMetadata && attributes && enrollment) {
+        const dataValueList = flatDataValuesFromEvents(enrollment.events);
+        const dataElements = getDataElementsFromProgram(programMetadata, dataValueList);
+
+        const trackedEntityAttributes = attributes.reduce((acc, item) => {
+            acc[item.attribute] = { id: item.attribute, valueType: item.valueType };
+            return acc;
+        }, {});
+
+        const teiAttributesValues = attributes?.reduce((acc, item) => {
+            acc[item.attribute] = item.value;
+            return acc;
+        }, {});
+
+        return runRulesForEnrollmentPage({
+            program,
+            orgUnit,
+            trackedEntityAttributes,
+            dataElementsInProgram: dataElements,
+            teiValues: teiAttributesValues,
+            eventsData: getEventsDataFromEnrollment(enrollment, dataElements),
+            enrollmentData: getEnrollmentData(enrollment),
+        });
+    }
+    return undefined;
```

Comment:
Hey, I am wondering if it makes sense to return something relevant here and maybe show it to the user. WHYT?

Reply:
Hey @simonadomnisoru, thanks for the comment. I think this should run in the background and not showing to the user. Maybe we can add a log for debugging purpose but I will let other team members decide @JoakimSM WDYT?

Reply:
I think the logic here is sound. The function returns undefined until we have all the required data.